cclib/parser/gamessdatparser.py

```diff
@@ -1,246 +1,314 @@
+        """Return a string representation of the object."""
+        return f"GAMESS .dat log file {self.filename}"
+
+    def __repr__(self):
+        """Return a representation of the object."""
+        return f'GAMESS .dat ("{self.filename}")'
+    
+    def normalisesym(self, label):
+        """Normalise the symmetries used by GAMESS .dat."""
+
+        pass
+
+    def before_parsing(self):
+        self.pt = PeriodicTable()
+        # To change: declared only for passing unit tests
+
+    def after_parsing(self):
+        if hasattr(self, "atomcoords"):
+            len_coords = len(self.atomcoords)
+            self.atomcoords = numpy.reshape(self.atomcoords, (1, len_coords, 3))
+        
+        if hasattr(self, "mocoeffs"):
+            self.mocoeffs = numpy.reshape(self.mocoeffs, (1, self.nmo, self.nbasis))
+
+
+    def extract(self, inputfile, line):
+        """Extract information from the file object inputfile."""
+        
+        # Extract element and its properties
+
+        # $DATA  
+        # water                                                                           
+        # C1       0
+        # O           8.0      0.0000000000      0.0000000000      0.0000000000
+        #    STO     3
+                
+        # H           1.0      0.9900000000      0.0000000000      0.0000000000
+        #    STO     3
+                
+        # H           1.0     -0.2728810000      0.9516490000      0.0000000000
+        #    STO     3
+                
+        #  $END   
+
+        # Extract molecule name
+
+        if line[1:6] == "$DATA":
+
+            line = next(inputfile)
+            name = line.split('\n')[0].strip()
+            self.metadata["name"] = name
+
+            # Extract atomic information
+
+            self.atommasses = []
+            self.atomcoords = []
+
+            line = next(inputfile)
+
+            while line.strip() != "$END":
+                parts = line.split()
+                if len(parts) > 2:
+
+                    symbol      = parts[0]
+                    mass        = float(parts[1])
+                    coordinates = [float(coord) for coord in parts[2:]]
+
+                    assert len(coordinates) == 3
+
+                    self.atommasses.append(mass)
+                    self.atomcoords.append(coordinates)
+                elif len(parts) == 2:
+                    basis_set = parts[0] + '-' + parts[1] + 'G'
+                    self.metadata['basis_set'] = basis_set
+
+                    
+                line = next(inputfile)
+
+
+        # Extract energy
+
+        # --- CLOSED SHELL ORBITALS --- GENERATED AT Mon Aug  5 13:05:47 2019
+        # water                                                                           
+        # E(RHF)=      -74.9643287920, E(NUC)=    8.8870072224,   13 ITERS
+
+        # Extract E(RHF) value
+
+        if "E(R" in line:
+            val_pattern = r"E\(R[^,]+"
+            match = re.search(val_pattern, line)

```

Comment:
I think this is from the previous PR, so apologies for missing it.  If you're going to use a regex here, which might actually be necessary later on, don't call the result `match`, since it's the name of a built-in keyword.  (A similar thing would be using `list` as a variable name.)  I see `mtch` sometimes but the name is up to you.